src/TrackPanel.h

```diff
@@ -224,4 +231,109 @@ struct AUDACITY_DLL_API IsVisibleTrack
    wxRect mPanelRect;
 };
 
+//! A class with context menu over audio track.
+class AUDACITY_DLL_API AudacityOverTrackContextMenu : public wxMenu
+{
+public:
+    enum MenuItemID
+    {
+        kItemID_Cut = wxID_LAST + 1,
+        kItemID_Copy,
+        kItemID_Paste,
+        kItemID_Split,
+        kItemID_Mute,
+        kItemID_Rename
+    };
+    
+    AudacityOverTrackContextMenu(CommandManager* commandManager)
+    {
+        auto cutItemString = XO("Cut");
+        auto copyItemString = XO("Copy");
+        auto pasteItemString = XO("Paste");
+        auto splitItemString = XO("Split clip");
+        auto muteItemString = XO("Mute/unmute track");
+        // TODO: When task #1258 will finished.
+        // For more info see #998 task description.
+        //auto renameClipItemString = XO("Rename clip");
+        
+        if (commandManager)
+        {
+            ComponentInterfaceSymbol cmdName;
+            
+            cmdName = ComponentInterfaceSymbol(wxT("Cut"), cutItemString);
+            cutItemString = commandManager->DescribeCommandsAndShortcuts(&cmdName, 1u);
+            
+            cmdName = ComponentInterfaceSymbol(wxT("Copy"), copyItemString);
+            copyItemString = commandManager->DescribeCommandsAndShortcuts(&cmdName, 1u);
+            
+            cmdName = ComponentInterfaceSymbol(wxT("Paste"), pasteItemString);
+            pasteItemString = commandManager->DescribeCommandsAndShortcuts(&cmdName, 1u);
+            
+            cmdName = ComponentInterfaceSymbol(wxT("Split"), splitItemString);
+            splitItemString = commandManager->DescribeCommandsAndShortcuts(&cmdName, 1u);
+            
+            cmdName = ComponentInterfaceSymbol(wxT("MuteTracks"), muteItemString);
+            muteItemString = commandManager->DescribeCommandsAndShortcuts(&cmdName, 1u);
+        }
+        
+        Append(MenuItemID::kItemID_Cut, cutItemString.Translation());
+        Append(MenuItemID::kItemID_Copy, copyItemString.Translation());
+        Append(MenuItemID::kItemID_Paste, pasteItemString.Translation());
+        AppendSeparator();
+        Append(MenuItemID::kItemID_Split, splitItemString.Translation());
+        Append(MenuItemID::kItemID_Mute, muteItemString.Translation());
```

Comment:
Again, make it table-driven.  Not repetitions calls to Append().

Reply:
Another problem is that the event dispatching code must specially cooperate with this menu construction code.  Much simpler just to have a table-driven way to do both things, and one common table top describe it.